generators/src/keyvalue.ts

```diff
@@ -23,58 +48,124 @@ export interface KeyValueListInit {
   delimiter?: string;
 
   /** List of key suggestions */
-  keys?: Suggestions;
+  keys?: KeyValueSuggestions;
 
   /** List of value suggestions */
-  values?: Suggestions;
+  values?: KeyValueSuggestions;
 
   /** Cache key and value suggestions */
-  cache?: boolean;
+  cache?: CacheValue;
+
+  /** Should the separator be inserted after a key? (default: true ) */
+  insertSeparator?: boolean;
+
+  /** Insert the delimiter string after accepting a value suggestion (default: false) */
+  insertDelimiter?: boolean;
 }
 
 /** Cache of Fig suggestions using the string[]/Suggestion[]/function as a key */
-const suggestionCache = new Map<Suggestions, Fig.Suggestion[]>();
+const suggestionCache = new Map<KeyValueSuggestions, Fig.Suggestion[]>();
+
+function appendToInsertValue(append: string, suggestions: Fig.Suggestion[]): Fig.Suggestion[] {
+  if (append.length === 0) {
+    return suggestions;
+  }
+  return suggestions.map((item) =>
+    item.insertValue ? item : { ...item, insertValue: item.name + append }
+  );
+}
 
 async function kvSuggestionsToFigSuggestions(
-  suggestions: Suggestions,
+  suggestions: KeyValueSuggestions,
+  append: string,
   init: Parameters<NonNullable<Fig.Generator["custom"]>>
 ): Promise<Fig.Suggestion[]> {
   if (typeof suggestions === "function") {
-    return suggestions(...init);
+    const out = await suggestions(...init);
+    return appendToInsertValue(append, out);
   }
   if (typeof suggestions[0] === "string") {
-    return (suggestions as string[]).map((name) => ({ name }));
+    const out = (suggestions as string[]).map((name) => ({ name }));
+    return appendToInsertValue(append, out);
   }
-  return suggestions as Fig.Suggestion[];
+  return appendToInsertValue(append, suggestions as Fig.Suggestion[]);
 }
 
 async function getSuggestions(
-  suggestions: Suggestions,
-  cache: boolean,
+  suggestions: KeyValueSuggestions,
+  append: string,
+  useSuggestionCache: boolean,
   init: Parameters<NonNullable<Fig.Generator["custom"]>>
 ): Promise<Fig.Suggestion[]> {
-  if (cache) {
+  if (useSuggestionCache || suggestions instanceof Array) {
```

Comment:
This may prevent some headaches in the future finding why the generator does not work in iframes😄
```suggestion
  if (useSuggestionCache || Array.isArray(suggestions)) {
```

Reply:
Ahh fair. I use `instanceof` by default because it's less overhead and TSC is more likely to be happy, but that's definitely a relevant concern